### DISCUSSION TITLE
Make frame limiter available for slower than the refresh rate

### DISCUSSION THREAD

**Discussion by Hinaxnitro** [ppy/osu#30760] [Ideas]
I'm playing on an Android phone, SHARP SH-M19.

This can hit 60fps (which is this one's refresh rate) sometimes while playing Mania
but soon the phone gets thermal throttled, causing freezes.
(changing to "Single thread" doesn't help)

I think limiting the Framerate to half of the refresh rate helps lighten this issue.
Since it can also reduce the Update rate that barely hits the limit
(weirdly Vsync setting doesn't help),
this can help make this game smoother.

**Comment by peppy** [comment]
Android devices generally have a "low power" mode – have you tried using that?

**Reply by Hinaxnitro** [reply]
I tried, but the issue didn't lighten.

This is my opinion but, I won't recommend Android's battery-saving feature as the solution.

These are the reasons:

Usually, this only throttles **background** activities so I cannot expect a performance throttling.

Also, what it does depends on the vendors.
Some do nothing special but others may do something like limiting the display framerate
(my older phone does but this phone doesn't because of no VRR)
and I think the apps can't expect what comes.

(this only works when the charger is unplugged but I don't want to play rhythm games with it unplugged is also the reason)

**Reply by peppy** [reply]
Fair. I can imagine a world where we have a 30 fps lock which is only available on mobile devices, if it comes to it.

But let's hope we can optimise things to the point this isn't required.
